### PR TITLE
Resolve options in AbstractElement on first access

### LINKED ISSUE #225: defer resolving admin-element options until first used
Currently options are resolved in constructor of [`AbstractElement`](https://github.com/fsi-open/admin-bundle/blob/master/Admin/AbstractElement.php#L32). This prevent for example building options based on resources added by workers (`DoctrineWorker` or `RequestStackWorker`).

Maybe options can be resolved at first use of: `getOption`, `getOptions`, `hasOption`?

### PR DESCRIPTION
**Pull request by szymach**
fix #225

### CODE REVIEW

File: Admin/AbstractElement.php [hunk 1]
```diff
@@ -20,16 +20,19 @@
...
+    /**
+     * @var array
+     */
+    private $unresolvedOptions;
 
     /**
      * @param array $options
      */
-    public function __construct($options = [])
+    public function __construct(array $options = [])
```
Comment (szymach) [inline Admin/AbstractElement.php]: Can this be an issue? I doubt anyone will inject `\ArrayAccess` objects, but still, should it stay or should I remove the typehint?

Comment (chives) [inline Admin/AbstractElement.php]: There was a typehint in docblock before so the class contract was the same (although not enforced by language)

**Comment by szymach** [comment]
@chives

**Comment by szymach** [comment]
@chives :)

**Pull request merged**